### PR TITLE
DEVOPS-1901 Fix error: az login again to refresh permissions

### PR DESCRIPTION
**Pull request by urbinaalex17**
## Type of change

```
- [X] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Split the remove docker images steps per registry since that's what causing the issue:

```bash
WARNING: Unable to get AAD authorization tokens with message: 2024-03-22 14:54:44.177986 An error occurred: CONNECTIVITY_REFRESH_TOKEN_ERROR
Access to registry 'bitwardenqa.azurecr.io' was denied. Response code: 403. Please try running 'az login' again to refresh permissions.
```

## Code changes

* **.github/workflows/cleanup-after-pr.yml:** Separate the removal of docker images per registry instead of using a list of registries.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: .github/workflows/cleanup-after-pr.yml [hunk 1]
```diff
@@ -19,6 +19,47 @@ jobs:
...
+            )
+
+            if [[ "$TAG_EXISTS" == "true" ]]; then
+              echo "[*] Tag exists. Removing tag"
+              az acr repository delete --name $REGISTRY --image $SERVICE_NAME:$IMAGE_TAG --yes
+            else
+              echo "[*] Tag does not exist. No action needed"
+            fi
+          done
+
```
Comment (mimartin12) [inline .github/workflows/cleanup-after-pr.yml:62]: This registry isn't even used, only `bitwardenprod` https://github.com/bitwarden/server/blob/5a5e5c50580055f3dd681cf763627b0324d73e82/.github/workflows/build.yml#L13-L14. I think this, along with other references, can be removed from this workflow. Feel free to handle what is outside of scope in a different change, but this specific one can be removed.

File: .github/workflows/cleanup-after-pr.yml [hunk 2]
```diff
@@ -28,13 +69,10 @@ jobs:
...
       ########## Remove Docker images ##########
-      - name: Remove the Docker image from ACR
+      - name: Remove the Docker image from bitwardenprod ACR
         env:
           REF: ${{ github.event.pull_request.head.ref }}
-          REGISTRIES: |
-            registries:
-              - bitwardenprod
-              - bitwardenqa
+          REGISTRY: bitwardenprod
```
Comment (mimartin12) [inline .github/workflows/cleanup-after-pr.yml:75]: 🌱  There are multiple references to `bitwardenprod` and some duplication, you could improve this by using a global environment variable. Example https://github.com/bitwarden/server/blob/5a5e5c50580055f3dd681cf763627b0324d73e82/.github/workflows/build.yml#L13-L14

**Review by mimartin12 — CHANGES_REQUESTED** [review]
Some of the fixes include references to a container registry that just isn't used, as shown in the build workflows. We don't need it and can remove.

**Comment by urbinaalex17** [comment]
@mimartin12 , is the qa registry the one that we are not using?

**Comment by mimartin12** [comment]
> @mimartin12 , is the qa registry the one that we are not using?

Yes, the build workflow only pushes to prod. https://github.com/bitwarden/server/blob/5a5e5c50580055f3dd681cf763627b0324d73e82/.github/workflows/build.yml#L13-L14

File: .github/workflows/cleanup-after-pr.yml (hunk 1 above)
Comment (urbinaalex17) [reply .github/workflows/cleanup-after-pr.yml]: Thanks for the suggestion!

File: .github/workflows/cleanup-after-pr.yml (hunk 2 above)
Comment (urbinaalex17) [reply .github/workflows/cleanup-after-pr.yml]: Suggestion addressed. Thanks, @mimartin12!

File: .github/workflows/cleanup-after-pr.yml [hunk 3]
```diff
@@ -5,36 +5,27 @@ on:
...
 
       ########## Remove Docker images ##########
       - name: Remove the Docker image from ACR
         env:
           REF: ${{ github.event.pull_request.head.ref }}
-          REGISTRIES: |
-            registries:
-              - bitwardenprod
-              - bitwardenqa
+          REGISTRY: bitwardenprod
```
Comment (mimartin12) [inline .github/workflows/cleanup-after-pr.yml:28]:
```suggestion
```
❓ Still needed?

**Review by mimartin12 — CHANGES_REQUESTED** [review]

**Review by mimartin12 — APPROVED** [review]

**Pull request merged**